serde_with/src/formats.rs

```diff
@@ -80,6 +80,59 @@ create_format!(
     Unpadded
 );
 
+/// When serializing a value of a type,
+///   that allows multiple types during deserialization,
+/// prefer a specific type.
+pub trait TypePreference: SerializeAs<[u8]> {}
```

Comment:
That's neat to just reuse the `SerializeAs` trait for the behavior.

Does the trait need to be fully public? We can't make it private, since it appears in public bounds, but seal all implementations. Similar to https://github.com/jonasbb/serde_with/blob/bc2063440da5168963285196e0345e82ab9ccc4e/serde_with/src/base64.rs#L160-L168.

I would like to keep the serialization and deserialization behavior matching if possible. For that a fully public trait feels unnecessary. Having ASCII and unicode strings supported feels sufficient for now. If it turns out that more freedom is necessary, it can be unsealed later.